bigtable/admin.go

```diff
@@ -376,79 +439,111 @@ func (ac *AdminClient) CreateColumnFamilyWithConfig(ctx context.Context, table,
+	automatedBackupPolicyFieldMask = "automated_backup_policy"
+)
 
-// updateTableWithConf updates a table in the instance from the given configuration.
-// only deletion protection can be updated at this period.
-// table ID is required.
-func (ac *AdminClient) updateTableWithConf(ctx context.Context, conf *UpdateTableConf) error {
-	if conf.tableID == "" {
-		return errors.New("TableID is required")
+func (ac *AdminClient) newUpdateTableRequestProto(tableID string) (*btapb.UpdateTableRequest, error) {
+	if tableID == "" {
+		return nil, errors.New("TableID is required")
 	}
-
-	ctx = mergeOutgoingMetadata(ctx, ac.md)
-
 	updateMask := &field_mask.FieldMask{
 		Paths: []string{},
 	}
-	prefix := ac.instancePrefix()
 	req := &btapb.UpdateTableRequest{
 		Table: &btapb.Table{
-			Name: prefix + "/tables/" + conf.tableID,
+			Name: ac.instancePrefix() + "/tables/" + tableID,
 		},
 		UpdateMask: updateMask,
 	}
+	return req, nil
+}
 
-	if conf.deletionProtection != None {
-		updateMask.Paths = append(updateMask.Paths, "deletion_protection")
-		req.Table.DeletionProtection = conf.deletionProtection != Unprotected
-	}
-
-	if conf.changeStreamRetention != nil {
-		if conf.changeStreamRetention.(time.Duration) == time.Duration(0) {
-			updateMask.Paths = append(updateMask.Paths, "change_stream_config")
-		} else {
-			updateMask.Paths = append(updateMask.Paths, "change_stream_config.retention_period")
-			req.Table.ChangeStreamConfig = &btapb.ChangeStreamConfig{}
-			req.Table.ChangeStreamConfig.RetentionPeriod = durationpb.New(conf.changeStreamRetention.(time.Duration))
-		}
-	}
+func (ac *AdminClient) updateTableAndWait(ctx context.Context, updateTableRequest *btapb.UpdateTableRequest) error {
+	ctx = mergeOutgoingMetadata(ctx, ac.md)
 
-	lro, err := ac.tClient.UpdateTable(ctx, req)
+	lro, err := ac.tClient.UpdateTable(ctx, updateTableRequest)
 	if err != nil {
 		return fmt.Errorf("error from update: %w", err)
 	}
+
 	var tbl btapb.Table
 	op := longrunning.InternalNewOperation(ac.lroClient, lro)
 	err = op.Wait(ctx, &tbl)
 	if err != nil {
 		return fmt.Errorf("error from operation: %v", err)
 	}
+
 	return nil
 }
 
+// UpdateTableDisableChangeStream updates a table to disable change stream for table ID.
+func (ac *AdminClient) UpdateTableDisableChangeStream(ctx context.Context, tableID string) error {
+	req, err := ac.newUpdateTableRequestProto(tableID)
+	if err != nil {
+		return err
+	}
+	req.UpdateMask.Paths = []string{changeStreamRetentionFieldMask}
+	return ac.updateTableAndWait(ctx, req)
+}
+
+// UpdateTableWithChangeStream updates a table to with the given table ID and change stream config.
+func (ac *AdminClient) UpdateTableWithChangeStream(ctx context.Context, tableID string, changeStreamRetention ChangeStreamRetention) error {
+	req, err := ac.newUpdateTableRequestProto(tableID)
+	if err != nil {
+		return err
+	}
+	req.UpdateMask.Paths = []string{changeStreamRetentionFieldMask + ".retention_period"}
```

Comment:
should th eupdate mask include change_stream_config?

Reply:
Good catch. Thanks. This was a poorly-named variable, I've now changed it to `changeStreamConfigFieldMask` (`change_stream_config` is still included)